src/manifest.rs

```diff
@@ -33,15 +33,212 @@ use crate::{
 /// The name of the manifest file
 pub const MANIFEST_FILE: &str = "Proto.toml";
 
-/// A `buffrs` manifest format used for serialization and deserialization.
+/// The canary edition supported by this version of buffrs
+pub const CANARY_EDITION: &str = concat!("0.", env!("CARGO_PKG_VERSION_MINOR"));
+
+/// Edition of the buffrs manifest
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Edition {
+    /// The canary edition of manifests
+    ///
+    /// This indicates that breaking changes and unstable behavior can occur
+    /// at any time. Users are responsible for consulting documentation and
+    /// help channels if errors occur.
+    Canary,
```

Comment:
This special-casing is what I worry about. What changes will be needed here once we release `v1.0`?

Reply:
`Stable` can then be explicitly be compatible with certain older editions that are bijective to the new structure

Reply:
Idea: I instead of `Stable` what about something like `1` to be able to handle a possible future edition `2` in a consistant manner. Am I correct?

In this line of thought what about renaming `Canary` to `Canary1`. Then if we introduce breaking change we would introduce `Canary2` and might just drop `Canary1` or keep compatiblity for some time with it?

Reply:
Yes ofc, the internal edition name (eg stable, or V1) would probably be serialized as edition = "1"